src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -326,16 +1026,37 @@ private static ulong DigitsToUInt64(byte* p, int count)
             Debug.Assert((1 <= count) && (count <= 19));
 
             byte* end = (p + count);
-            ulong res = (ulong)(p[0] - '0');
+            ulong res = 0;
+
+            // parse batches of 8 digits with SWAR
+            while (end - p >= 8)
+            {
+                res = (res * 100000000) + ParseEightDigitsUnrolled(p);
+                p += 8;
+            }
 
-            for (p++; p < end; p++)
+            while (p != end)
             {
                 res = (10 * res) + p[0] - '0';
+                ++p;
             }
 
             return res;
         }
 
+        /// <summary>Parse eight consecutive digits using SWAR</summary>
+        internal static uint ParseEightDigitsUnrolled(byte* chars)
```

Comment:
FYI: this method is not inlined at the moment, probably it's worth to add [AggressiveInlining]
![image](https://user-images.githubusercontent.com/523221/145720070-996ae57c-68cf-4ab9-b96e-f23c0c002847.png)

Reply:
fine, i'll do it !